lib/libshare/smb.c

```diff
@@ -153,39 +157,48 @@ smb_retrieve_shares(void)
 					goto out;
 				}
 
-				strncpy(shares->name, name,
+				strlcpy(shares->name, name,
 					sizeof (shares->name));
```

Comment:
cstyle - four space indent for continued lines.  This same fix needs to made a few places in this function.